### PR TITLE
Automated cherry pick of #58438: fix apiserver crash caused by nil pointer and ensure CRD

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Cherry pick of #58438 on release-1.9.

#58438: fix apiserver crash caused by nil pointer and ensure CRD

```release-note
CustomResourceDefinitions: OpenAPI v3 validation schemas containing `$ref`references are no longer permitted. Before upgrading, ensure CRD definitions do not include those `$ref` fields.
```

### CODE REVIEW

**Comment by nikhita** [comment]
> r.getOrCreateServingInfoFor undefined (type *crdHandler has no field or method getOrCreateServingInfoFor)

@hzxuzhonghu I'm guessing there should have been conflicts and something went wrong here?

**Comment by carlory** [comment]
@hzxuzhonghu 

getOrCreateServingInfoFor -> getServingInfoFor

https://github.com/kubernetes/kubernetes/blob/v1.9.0/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L281

**Comment by hzxuzhonghu** [comment]
Yes， it did conflicts. And I make a mistake when resolve it.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go [hunk 1]
```diff
@@ -276,13 +273,13 @@ func (r *crdHandler) removeDeadStorage() {
 }
 
 // GetCustomResourceListerCollectionDeleter returns the ListerCollectionDeleter for
-// the given uid, or nil if one does not exist.
-func (r *crdHandler) GetCustomResourceListerCollectionDeleter(crd *apiextensions.CustomResourceDefinition) finalizer.ListerCollectionDeleter {
+// the given uid, or nil if an error occurs.
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go]: the "or nil ..." is not needed here. We have the error result type now (the semantics is standard, no need to document it)

**Comment by sttts** [comment]
One nit, otherwise lgtm (Feel free to self-apply).

**Comment by mbohlool** [comment]
The release note on the parent PR is long. Can you please make it brief yet complete?

**Comment by mbohlool** [comment]
@sttts is your comment specific to release-1.9 branch? if not, it should be fixed in the parent PR not here.

**Comment by hzxuzhonghu** [comment]
@mbohlool I will address this comments, and I think it's applied to both 1.9 and master

**Comment by hzxuzhonghu** [comment]
/release-note CustomResourceDefinitions: OpenAPI v3 validation schemas containing `$ref`references are no longer permitted. Before upgrading, ensure CRD definitions do not include those `$ref` fields.

**Comment by hzxuzhonghu** [comment]
```release-note
CustomResourceDefinitions: OpenAPI v3 validation schemas containing `$ref`references are no longer permitted. Before upgrading, ensure CRD definitions do not include those `$ref` fields.
```

**Comment by hzxuzhonghu** [comment]
@nikhita for lgtm

**Comment by mbohlool** [comment]
@hzxuzhonghu please fix it in master and cherrypick both together using cherrypick tool again. thanks.

**Comment by hzxuzhonghu** [comment]
ok

**Comment by nikhita** [comment]
/assign
/reopen

@hzxuzhonghu Can you cherry pick https://github.com/kubernetes/kubernetes/pull/58914 too?

**Comment by nikhita** [comment]
/reopen

@k8s-ci-robot :(

**Comment by hzxuzhonghu** [comment]
@nikhita ok, will do later.

**Comment by mbohlool** [comment]
@hzxuzhonghu 
https://github.com/kubernetes/kubernetes/pull/58914 is not in this cherry pick. You can create a new batch cherrypick for both PRs but I am ok if you just apply it manually as it is only a comment change. do it as a separate commit please and reference 58914.

**Comment by hzxuzhonghu** [comment]
ok

**Comment by hzxuzhonghu** [comment]
#58438 opened so close this.

**Pull request closed without merging**